middleware/response_test.go

```diff
@@ -31,12 +31,50 @@ func TestBadResponseLoggingWriter(t *testing.T) {
 		default:
 			http.Error(wrapped, tc.data, tc.statusCode)
 		}
-		if wrapped.statusCode != tc.statusCode {
-			t.Errorf("Wrong status code: have %d want %d", wrapped.statusCode, tc.statusCode)
+		if wrapped.getStatusCode() != tc.statusCode {
+			t.Errorf("Wrong status code: have %d want %d", wrapped.getStatusCode(), tc.statusCode)
 		}
 		data := string(buf.Bytes())
 		if data != tc.expected {
 			t.Errorf("Wrong data: have %q want %q", data, tc.expected)
 		}
 	}
 }
+
+// nonFlushingResponseWriter implements http.ResponseWriter but does not implement http.Flusher
+type nonFlushingResponseWriter struct{}
+
+func (rw *nonFlushingResponseWriter) Header() http.Header {
+	return nil
+}
+
+func (rw *nonFlushingResponseWriter) Write(_ []byte) (int, error) {
+	return -1, nil
+}
+
+func (rw *nonFlushingResponseWriter) WriteHeader(_ int) {
+}
+
+func TestNonFlushingBadResponseLoggingWriter(t *testing.T) {
```

Comment:
[nit] This test both cases, so we could rename it. For example:

```suggestion
func TestBadResponseLoggingWriter_WithAndWithoutFlusher(t *testing.T) {
```